app/controllers/hub/coalitions_controller.rb

```diff
@@ -0,0 +1,43 @@
+module Hub
+  class CoalitionsController < ApplicationController
+    include AccessControllable
+    before_action :require_sign_in
+    load_and_authorize_resource :coalition, parent: false
+
+    layout "hub"
+
+    def new; end
+
+    def create
+      @form = CoalitionForm.new(coalition_params.merge(state_routing_target_params).merge({ coalition: @coalition }))
```

Comment:
Trivial nit:

Since you're asking cancancan to `load_and_authorize_resource` with `:coalition`, it will already call `coalition_params` as part of creating `@coalition`. Given that, I wonder if we can simplify the form to not take `name` as param, since it'll be part of the `@coalition` already. In that case, it might look like:

```
CoalitionForm.new({coalition: @coalition}.merge(state_routing_target_params))
```

This would allow us to remove `@coalition.name = name` from the form, too.

Another form that I would prefer even more would be:

```
CoalitionForm.new(@coalition, state_routing_target_params[:states])
```

Reply:
I'll second the suggestion to send the params in as separate arguments! It seems much clearer in terms of looking at what the form is going to be responsible for.